src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -428,13 +428,30 @@ p2ptest_set_or_cmp(char *boptr, size_t size, char pattern, bool set)
 {
   int stride = xrt_core::getpagesize();
 
+  // Generate the valid data vector
+  const size_t write_size = 1024;
+  char valid_data[write_size];
+  std::memset(valid_data, pattern, write_size);
+
   assert((size % stride) == 0);
   for (size_t i = 0; i < size; i += stride) {
-    if (set) {
-      boptr[i] = pattern;
-    }
-    else if (boptr[i] != pattern) {
-      return false;
+    // Due to test timeout we can only have one large write to test write combines
+    // The first write in a memory segment is a large write combine while all others are single bytes
+    if (set)
+      if(i == 0)
```

Comment:
This code is screaming to be its own method for 1 flow is using a character (e.g., pattern) to initialize the memory and the other a "buffer" of data.

Yes, in this case it is the same character, but this code should be refactor to separate the difference algorithm to allow for future different memory pattern blocks.

Reply:
Let me know if you like the changes!